bayesian/api/api_v2.py

```diff
@@ -202,6 +203,63 @@ def get(ecosystem, package, version):
 
         raise HTTPError(404, msg)
 
+    @staticmethod
+    def post():
+        """Handle the POST REST API call.
+
+        Component Analyses Batch is a 4 Step Process:
+        1. Gather and clean Request.
+        2. Query GraphDB Vendor Specific Edge
+        3. Build Known Package Response and Trigger componentApiFlow
+        4. Handle Unknown Packages and Trigger bayesianApiFlow
+        """
+        response_template = namedtuple("response_template", ["message", "status"])
+        ingestion_disabled_msg: str = "No data found for any package in manifest file. " \
+                                      "Ingestion flow skipped as DISABLE_UNKNOWN_PACKAGE_FLOW " \
+                                      "is enabled"
+        no_package_available_msg: str = "No Package in given manifest is available. " \
+                                        "Packages will be available shortly, " \
+                                        "Please retry after some time."
+
+        # Step1: Gather and clean Request
+        input_json: dict = request.get_json()
+        ecosystem: str = input_json.get('ecosystem')
+        try:
+            packages_list = ca_validate_input(input_json, ecosystem)
+        except BadRequest as br:
+            logger.debug(br)
+            raise HTTPError(400, str(br))
```

Comment:
Use `raise YourException from e` pattern everywhere.
```suggestion
            raise HTTPError(400, str(br)) from br
```

Reply:
Why..? Whats the Gain?

Reply:
It retains the original exception call trace, without that you will lose the stack trace which originally raised the exception.

Reply:
Ack!